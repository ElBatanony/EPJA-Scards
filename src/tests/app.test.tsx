import App from "../app";
import React from "react";
import { mount } from "enzyme";

import { initMock, makeRequest } from "./mockUtils";

import {
  initResponses,
  nextMessageResponse,
  previousMessageResonse,
  scardResponse,
  studyNotesLoadResponse,
  studyNotesSaveResponse,
  updatedScardResponse,
  updatedScardsResponse,
} from "./mockResponses";

describe("App Test", () => {
  let mock;

  beforeEach(() => {
    mock = initMock();
  });

  it("Render and test app", async () => {
    const app = mount(<App />);

    await makeRequest(app, mock, initResponses);

    // Inital state check
    expect(app.find("Text#welcomeToScards")).toMatchSnapshot();
    expect(app.find("Text#MsgText")).toMatchSnapshot();
    expect(app.find(".ScardQ")).toMatchSnapshot();
    expect(app.find("textarea#StudyNotesTextField")).toMatchSnapshot();
    expect(app.find(".ScardCard")).toHaveLength(5);

    // Clicking on Next Message
    app.find("button#NextMsgBtn").simulate("click");
    await makeRequest(app, mock, nextMessageResponse);
    expect(app.find("Text#MsgText")).toMatchSnapshot();

    // Clicking on Previous Message
    app.find("button#PrevMsgBtn").simulate("click");
    await makeRequest(app, mock, previousMessageResonse);
    expect(app.find("Text#MsgText")).toMatchSnapshot();

    // Loading, modifying and saving study notes
    app.find("button#LoadStudyNotes").simulate("click");
    await makeRequest(app, mock, studyNotesLoadResponse);
    app
      .find("textarea#StudyNotesTextField")
      .simulate("change", { target: { value: "Hello Study Notes" } });
    app.find("button#SaveStudyNotes").simulate("click");
    await makeRequest(app, mock, studyNotesSaveResponse);
    expect(app.find("textarea#StudyNotesTextField")).toMatchSnapshot();

    // Selecting the first scard
    app.find(".ScardCard").at(0).find('Stack[role="button"]').simulate("click");
    await makeRequest(app, mock, scardResponse);
    expect(app.find(".ScardCard")).toHaveLength(1);

    // Edit Scard
    app.find("button#EditScardBtn").simulate("click");
    app.update();
    expect(
      app.find("#ScardDialog").find("input#newScardQuestion")
    ).toMatchSnapshot();
    app
      .find("input#newScardQuestion")
      .simulate("change", { target: { value: "Question 1 updated" } });
    app.update();
    app.find("button#ScardDialogActionBtn").simulate("click");
    await makeRequest(app, mock, [
      updatedScardsResponse[0],
      updatedScardResponse[0],
    ]);
    expect(app.find(".ScardQ")).toMatchSnapshot();
  });
});
