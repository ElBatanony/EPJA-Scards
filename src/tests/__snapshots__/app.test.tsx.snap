// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Render and test app 1`] = `
<Text
  id="welcomeToScards"
  key=".0:$.0"
  variant="superLarge"
>
  <span
    className="css-42"
    id="welcomeToScards"
  >
    Welcome to Scards!
  </span>
</Text>
`;

exports[`App Test Render and test app 2`] = `
<Text
  id="MsgText"
  key=".0:$.$.0"
  variant="large"
>
  <span
    className="css-43"
    id="MsgText"
  >
    Welcome! This is message1
  </span>
</Text>
`;

exports[`App Test Render and test app 3`] = `
Array [
  <Text
    className="ScardQ"
    key=".0:$.$.0=020"
    variant="xLarge"
  >
    <span
      className="ScardQ css-71"
    >
      Question 1 updated
    </span>
  </Text>,
  <span
    className="ScardQ css-71"
  >
    Question 1 updated
  </span>,
  <Text
    className="ScardQ"
    key=".0:$.$.0=020"
    variant="xLarge"
  >
    <span
      className="ScardQ css-71"
    >
      Question 2
    </span>
  </Text>,
  <span
    className="ScardQ css-71"
  >
    Question 2
  </span>,
  <Text
    className="ScardQ"
    key=".0:$.$.0=020"
    variant="xLarge"
  >
    <span
      className="ScardQ css-71"
    >
      Question 3
    </span>
  </Text>,
  <span
    className="ScardQ css-71"
  >
    Question 3
  </span>,
  <Text
    className="ScardQ"
    key=".0:$.$.0=020"
    variant="xLarge"
  >
    <span
      className="ScardQ css-71"
    >
      Question 4
    </span>
  </Text>,
  <span
    className="ScardQ css-71"
  >
    Question 4
  </span>,
  <Text
    className="ScardQ"
    key=".0:$.$.0=020"
    variant="xLarge"
  >
    <span
      className="ScardQ css-71"
    >
      Question 5
    </span>
  </Text>,
  <span
    className="ScardQ css-71"
  >
    Question 5
  </span>,
]
`;

exports[`App Test Render and test app 4`] = `
<textarea
  aria-invalid={false}
  className="ms-TextField-field field-75"
  id="StudyNotesTextField"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  value=""
/>
`;

exports[`App Test Render and test app 5`] = `
<Text
  id="MsgText"
  key=".0:$.$.0"
  variant="large"
>
  <span
    className="css-43"
    id="MsgText"
  >
    Hello there! This is 2
  </span>
</Text>
`;

exports[`App Test Render and test app 6`] = `
<Text
  id="MsgText"
  key=".0:$.$.0"
  variant="large"
>
  <span
    className="css-43"
    id="MsgText"
  >
    Welcome! This is message1
  </span>
</Text>
`;

exports[`App Test Render and test app 7`] = `
<textarea
  aria-invalid={false}
  className="ms-TextField-field field-75"
  id="StudyNotesTextField"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  value="Hello Study Notes"
/>
`;

exports[`App Test Render and test app 8`] = `
<input
  aria-invalid={false}
  aria-labelledby="TextFieldLabel43"
  className="ms-TextField-field field-108"
  id="newScardQuestion"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  type="text"
  value="Question 1 updated"
/>
`;

exports[`App Test Render and test app 9`] = `
Array [
  <Text
    className="ScardQ"
    key=".0:$.$.0=020"
    variant="xLarge"
  >
    <span
      className="ScardQ css-71"
    >
      Question 1 updated
    </span>
  </Text>,
  <span
    className="ScardQ css-71"
  >
    Question 1 updated
  </span>,
]
`;
